jujugui/static/gui/src/app/components/user-profile/test-user-profile.js

```diff
@@ -750,4 +758,41 @@ describe('UserProfile', () => {
     assert.equal(charmstoreAbort.callCount, 2);
     assert.equal(listEnvsAbort.callCount, 1);
   });
+
+  it('can reset the model connection', () => {
+    var pluralize = sinon.stub();
+    pluralize.withArgs('model', sinon.match.any).returns('models');
+    pluralize.withArgs('bundle', sinon.match.any).returns('bundles');
+    pluralize.withArgs('charm', sinon.match.any).returns('charms');
+    var switchModel = sinon.stub();
+    var dbEnvironmentSet = sinon.stub();
+    var changeState = sinon.stub();
+    var renderer = jsTestUtils.shallowRender(
+      <juju.components.UserProfile
+        switchModel={switchModel}
+        users={users}
+        listEnvs={sinon.stub()}
+        showConnectingMask={sinon.stub()}
+        changeState={changeState}
+        charmstore={sinon.stub()}
+        dbEnvironmentSet={dbEnvironmentSet}
+        env={env}
+        getDiagramURL={sinon.stub()}
+        interactiveLogin={true}
+        storeUser={sinon.stub()}
+        pluralize={pluralize}
+        user={users.charmstore} />, true);
+    var instance = renderer.getMountedInstance();
+    instance.switchModel();
+    assert.equal(switchModel.callCount, 1,
```

Comment:
because the params that this `utilsSwitchModel` is called with are very important they should be checked here. Same goes for the `dbEnvironmentSet` call below.